### PR TITLE
LIME-475 - Third party API logic wrapped in conditional parameter.

### PR DESCRIPTION
**Pull request by MMcC-GDS**
### What changed

- Third party API was wrapped in a conditional parameter for later refactoring.

### Issue tracking

- [LIME-475](https://govukverify.atlassian.net/browse/LIME-475)

[LIME-475]: https://govukverify.atlassian.net/browse/LIME-475?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

With current third party API:
image

Without:
image

### CODE REVIEW

File: infrastructure/lambda/template.yaml [hunk 1]
```diff
@@ -103,6 +103,14 @@ Mappings:
       integration: 2
       production: 2
 
+  ThirdPartyApiMapping:
```
Comment (gds-jbentham) [inline infrastructure/lambda/template.yaml:106]: @MMcC-GDS Could we make this DVADigitalEnabled that way we can do a combination of if DVADigitalEnabled and header present use DVADigital else use existing.

Just trying to ensure we can enable and test the new method without blocking the old on the environments

File: lambdas/checkpassport/src/main/java/uk/gov/di/ipv/cri/passport/checkpassport/handler/CheckPassportHandler.java [hunk 2]
```diff
@@ -269,7 +270,7 @@ private APIGatewayProxyResponseEvent lambdaCompletedOK(
 
     private PassportFormData parsePassportFormRequest(String input)
             throws OAuthHttpResponseExceptionWithErrorBody {
-        LOGGER.info("Parsing passport form data into payload for DCS");
+        LOGGER.info("Parsing passport form data into payload");
```
Comment (gds-jbentham) [inline lambdas/checkpassport/src/main/java/uk/gov/di/ipv/cri/passport/checkpassport/handler/CheckPassportHandler.java:273]: Could this be for third party document check 

Just keeps it really clear

Comment (MMcC-GDS) [reply lambdas/checkpassport/src/main/java/uk/gov/di/ipv/cri/passport/checkpassport/handler/CheckPassportHandler.java]: Reworded.

File: infrastructure/lambda/template.yaml (hunk 1 above)
Comment (MMcC-GDS) [reply infrastructure/lambda/template.yaml]: Refactored and updated related fields and values.

File: lambdas/checkpassport/src/main/java/uk/gov/di/ipv/cri/passport/checkpassport/handler/CheckPassportHandler.java [hunk 3]
```diff
@@ -327,12 +328,20 @@ private DocumentDataVerificationService constructDocumentDataVerificationService
...
-                new ThirdPartyAPIService(
-                        passportConfigurationService,
-                        eventProbe,
-                        new DcsCryptographyService(passportConfigurationService),
-                        httpClient);
+        String thirdPartApiCheck =
+                passportConfigurationService.getParameterValue(THIRD_PARTY_API_CHECK);
+        LOGGER.info(String.format("Selecting %s as third party API", thirdPartApiCheck));
+        ThirdPartyAPIService thirdPartyAPIService = null;
+        if (thirdPartApiCheck.equals("DVADigital")) {
```
Comment (smsgds) [inline lambdas/checkpassport/src/main/java/uk/gov/di/ipv/cri/passport/checkpassport/handler/CheckPassportHandler.java:335]: !equals

**Review by gds-jbentham — APPROVED** [review]

**Pull request merged**